api/renterhost_test.go

```diff
@@ -2171,6 +2172,133 @@ func TestRedundancyReporting(t *testing.T) {
+	// blockchain.
+	err = fullyConnectNodes(testGroup)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Make sure that every wallet has money in it.
+	err = fundAllNodes(testGroup)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Add storage to every host.
+	err = addStorageToAllHosts(testGroup)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Announce every host.
+	err = announceAllHosts(testGroup)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Set an allowance with two hosts.
+	allowanceValues := url.Values{}
+	allowanceValues.Set("funds", "50000000000000000000000000000") // 50k SC
+	allowanceValues.Set("hosts", "2")
+	allowanceValues.Set("period", "10")
+	err = st.stdPostAPI("/renter", allowanceValues)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Block until the allowance has finished forming contracts.
+	err = build.Retry(50, time.Millisecond*250, func() error {
+		var rc RenterContracts
+		err = st.getAPI("/renter/contracts", &rc)
+		if err != nil {
+			return errors.New("couldn't get renter stats")
+		}
+		if len(rc.Contracts) != 2 {
+			return errors.New("no contracts")
+		}
+		return nil
+	})
+	if err != nil {
+		t.Fatal("allowance setting failed")
+	}
+
+	// Create a file to upload.
+	filesize := int(modules.SectorSize * 2)
+	path := filepath.Join(st.dir, "test.dat")
+	err = createRandFile(path, filesize)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Upload the file
+	dataPieces := 1
+	parityPieces := 1
+	uploadValues := url.Values{}
+	uploadValues.Set("source", path)
+	uploadValues.Set("datapieces", fmt.Sprint(dataPieces))
+	uploadValues.Set("paritypieces", fmt.Sprint(parityPieces))
+	err = st.stdPostAPI("/renter/upload/test", uploadValues)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Calculate the encrypted size of our fully redundant encoded file
+	pieceSize := modules.SectorSize - crypto.TwofishOverhead
+	chunkSize := pieceSize * uint64(dataPieces)
+	numChunks := uint64(filesize) / chunkSize
+	if uint64(filesize)%chunkSize != 0 {
+		numChunks++
+	}
+	fullyRedundantSize := modules.SectorSize * uint64(dataPieces+parityPieces) * uint64(numChunks)
+
+	// Monitor the file as it uploads. Ensure that the UploadProgress times
+	// the fully redundant file size always equals UploadedBytes reported
+	var rf RenterFiles
+	for i := 0; i < 60 && (len(rf.Files) != 1 || rf.Files[0].UploadProgress < 100); i++ {
+		st.getAPI("/renter/files", &rf)
+		if len(rf.Files) >= 1 {
+			uploadProgressBytes := uint64(float64(fullyRedundantSize) * rf.Files[0].UploadProgress / 100.0)
+			// Note: in Go 1.10 we will be able to write Math.Round(uploadProgressBytes) != rf.Files[0].UploadedBytes
+			if uploadProgressBytes != rf.Files[0].UploadedBytes && (uploadProgressBytes+1) != rf.Files[0].UploadedBytes {
+				t.Fatal(fmt.Sprintf("api reports having uploaded %v bytes when upload progress is %v%%, but the actual uploaded bytes count should be %v", rf.Files[0].UploadedBytes, rf.Files[0].UploadProgress, uploadProgressBytes))
```

Comment:
You can use `t.Fatalf("formatstring", args...)` to format the string instead of `t.Fatal(fmt.Sprintf("formatstring", args...)`. I would also add a newline after the format string and split that call over 2 lines.